blobs/howto/python/python-v12/crud_datalake.py

```diff
@@ -24,287 +24,180 @@
+    # -------------------------------------------------
+    # Create directory method
+    # -------------------------------------------------
 
-# -------------------------------------------------
-# Rename directory method
-# ------------------------------------------------
- 
-# <Snippet_RenameDirectory>
-def rename_directory():
-    try:
-       
-       file_system_client = service_client.get_file_system_client(file_system="my-file-system")
-       directory_client = file_system_client.get_directory_client("my-directory")
-       
-       new_dir_name = "my-directory-renamed"
-       directory_client.rename_directory(new_name=directory_client.file_system_name + '/' + new_dir_name)
-
-    except Exception as e:
-     print(e) 
-# </Snippet_RenameDirectory>
+    # <Snippet_CreateDirectory>
+    def create_directory(self, file_system_client: FileSystemClient, directory_name: str) -> DataLakeDirectoryClient:
+        directory_client = file_system_client.create_directory(directory_name)
 
-# -------------------------------------------------
-# Delete directory method
-# -------------------------------------------------
-
-# <Snippet_DeleteDirectory>
-def delete_directory():
-    try:
-        file_system_client = service_client.get_file_system_client(file_system="my-file-system")
-        directory_client = file_system_client.get_directory_client("my-directory")
+        return directory_client
+    # </Snippet_CreateDirectory>
 
+    # -------------------------------------------------
+    # Rename directory method
+    # ------------------------------------------------
+    
+    # <Snippet_RenameDirectory>
+    def rename_directory(self, directory_client: DataLakeDirectoryClient, new_dir_name: str):
+        directory_client.rename_directory(
+            new_name=f"{directory_client.file_system_name}/{new_dir_name}")
+    # </Snippet_RenameDirectory>
+
+    # -------------------------------------------------
+    # Delete directory method
+    # -------------------------------------------------
+
+    # <Snippet_DeleteDirectory>
+    def delete_directory(self, directory_client: DataLakeDirectoryClient):
         directory_client.delete_directory()
-    except Exception as e:
-     print(e) 
-# </Snippet_DeleteDirectory>
-
-# ------------------------------------------------
-# Upload a file to a directory
-# ------------------------------------------------
-
-# <Snippet_UploadFile>
-def upload_file_to_directory():
-    try:
+    # </Snippet_DeleteDirectory>
 
-        file_system_client = service_client.get_file_system_client(file_system="my-file-system")
+    # ------------------------------------------------
+    # Upload a file to a directory
+    # ------------------------------------------------
 
-        directory_client = file_system_client.get_directory_client("my-directory")
+    # <Snippet_UploadFile>
+    def upload_file_to_directory(self, directory_client: DataLakeDirectoryClient, local_path: str, file_name: str):
```

Comment:
.NET reference says that append is limited to 4000MiB per request. I assume this is true for Python as well? That is, this is a service limitation and not a client library limitation, correct?

Reply:
To be honest, I'm not sure what the limit is, and it doesn't seem to be documented in the REST doc. I would assume .NET has it right though and that would be the same for Python as it is controlled by the service. There are no client-side limits in place in Python at least.